### PR TITLE
utils_misc: Add function get_pid()

### PR DESCRIPTION
**Pull request by hahakiki2010**
Add this common operation in avocado-vt, consider
it maybe useful for others

Signed-off-by: Yan Li <yannli@redhat.com>

### CODE REVIEW

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -3419,3 +3419,17 @@ def check_machine_type_arch(machine_type):
                                        "supported on the host with "
                                        "arch '%s'" % (machine_type,
                                                       arch))
+
+
+def get_libvirtd_pid():
```
Comment (balamuruhans) [inline virttest/utils_test/libvirt.py]: we can place it in utils_misc and tt can be something like,

```python
def get_pid(process, session=None):
    """
    :param process: name of the process to retrieve its pid
    :param session: ShellSession object of VM or remote host
    :return:  pid of the process or return None if process is not available
    """
```

**Comment by hahakiki2010** [comment]
@balamuruhans I updated the func to be more general. Pls help review again. Thanks

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -4538,3 +4538,23 @@ def _wait_for_commands(bg_jobs, start_time, timeout):
...
+
+def get_pid(name, session=None):
+    """
+    Get pid by process name
+
+    :param name: Name of the process to retrieve its pid
+    :param session: ShellSession object of VM or remote host
+    :return: Pid of the process or raise exception if process is not available
+    """
+    cmd = "ps -ef | grep %s | grep -v grep" % name
```
Comment (balamuruhans) [inline virttest/utils_misc.py]: let us wrap the name in quotes,
`cmd = "ps -ef | grep '%s' | grep -v grep" % name`

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -4538,3 +4538,23 @@ def _wait_for_commands(bg_jobs, start_time, timeout):
...
+    if session:
+        status, output = session.cmd_status_output(cmd)
+    else:
+        ret = process.run(cmd, shell=True, ignore_status=True)
+        status, output = ret.exit_status, results_stdout_52lts(ret).strip()
+    if status:
+        raise exceptions.TestError("Pls check if %s is available" % name)
```
Comment (balamuruhans) [inline virttest/utils_misc.py]: IMO, I think returning None would be better. if test needs to check for a negative scenario and proceed then returning None would help.

File: virttest/utils_misc.py (hunk 2 above)
Comment (hahakiki2010) [reply virttest/utils_misc.py]: fixed

File: virttest/utils_misc.py (hunk 3 above)
Comment (hahakiki2010) [reply virttest/utils_misc.py]: agreed and updated

**Comment by hahakiki2010** [comment]
@balamuruhans All comments have updated. Thanks!

**Comment by balamuruhans** [comment]
Thanks @hahakiki2010 @sathnaga. let's merge it.

**Pull request merged**